frontend/online-shop/src/components/ProductDetailedView.tsx

```diff
@@ -0,0 +1,123 @@
+import { Typography, Button } from "@material-ui/core";
+import Grid from "@material-ui/core/Grid/Grid";
+import { useParams, Link } from "react-router-dom";
+import { useProductDetailViewStyles } from "../styles/ProductDetailViewStyle";
+
+let theProducts = [
```

Comment:
You should extract those either in a separate file or in the product list file for now

Reply:
Actually I noticed that you have them both in the list and in here. My recommendation would be to create a `productService` in which you declare this list as a `const` and also create two functions: `getAllProducts` and `getProductById`. From there you know what to change. At chapter 8 you will change the implementations for those two functions by calling your backend and returning the respective results.